dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/inbound/HttpInputStreamImpl.java

```diff
@@ -76,9 +84,58 @@ public HttpInputStreamImpl(HttpInboundServiceContext context, FullHttpRequest re
         this.nettyRequest = request;
         this.nettyBody = nettyRequest.content();
         buffer = ChannelFrameworkFactory.getBufferManager().wrap(nettyBody.nioBuffer()).position(nettyBody.readerIndex());
+        // Check if the request content is compressed
```

Comment:
This will be tried as an independent handler before the HTTP Dispatcher Handler. For now it will auto decompress when the transport receives the full request.